tool/tsh/tsh.go

```diff
@@ -578,6 +578,10 @@ func onLogin(cf *CLIConf) {
 		utils.FatalError(err)
 	}
 
+	// the login operation may update the username and should be considered the more
+	// "authoritative" source.
+	cf.Username = tc.Username
```

Comment:
nit: I understand this is a backport but resetting the usernname makes the above username assignment redundant.